filters/auth/oidc.go

```diff
@@ -660,6 +690,122 @@ func (f *tokenOidcFilter) callbackEndpoint(ctx filters.FilterContext) {
 	f.doDownstreamRedirect(ctx, encryptedData, f.getMaxAge(claimsMap), oauthState.RedirectUrl)
 }
 
+// handleDistributedClaims handles if user has an distributed / overage token.
+// See https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#groups-overage-claim
+// In Azure, if you are indirectly member of more than 200 groups, they will
+// send _claim_names and _claim_sources instead of the groups, per OIDC Core 1.0, section 5.6.2:
+// https://openid.net/specs/openid-connect-core-1_0.html#AggregatedDistributedClaims
+// This could be used with other providers as well. Example:
+//
+// {
+// 	 "_claim_names": {
+// 	   "groups": "src1"
+// 	 },
+// 	 "_claim_sources": {
+// 	   "src1": {
+// 	     "endpoint": "https://graph.windows.net/.../getMemberObjects"
+// 	   }
+//   }
+// }
+//
+func (f *tokenOidcFilter) handleDistributedClaims(ctx context.Context, oidcIDToken string, oauth2Token *oauth2.Token, claimsMap map[string]interface{}) error {
+	idToken, err := f.verifier.Verify(ctx, oidcIDToken)
+	if err != nil {
+		return fmt.Errorf("failed to verify id token: %w", err)
+	}
+
+	//https://github.com/coreos/go-oidc/issues/171#issuecomment-1044286153
+	var distributedClaims overageClaims
+	err = idToken.Claims(&distributedClaims)
+	// successful unmarshaling means we do have a case of distributed claims
+	if err != nil {
+		log.Info("No distributed claims found")
+		log.Debug(err)
+		return nil
+	}
+	claimMapSources := make(map[string]claimSource, len(distributedClaims.ClaimNames))
+	for claim, pointer := range distributedClaims.ClaimNames {
+		claimMapSources[claim] = distributedClaims.ClaimSources[pointer]
+	}
+
+	// TODO: what options should we implement?
+	// net.Options{
+	client := snet.NewClient(snet.Options{IdleConnTimeout: 2 * time.Second})
```

Comment:
client should be part of the filter or re-use authClient, if possible

Reply:
By using the current access token for the (recursive) calls we'll have only a one time usage per URI + token AFAIK. I believe we would clash creds if we store the authClient per URI!?

Reply:
yeah or just add the client to the filter such that you have it already instantiated.

Reply:
implemented